README.md

```diff
@@ -1 +1,71 @@
-# azure-cosmosdb-mongodbshardedcluster
\ No newline at end of file
+# Install MongoDB Sharding Cluster
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fcjsingh8512%2Fazure-cosmosdb-mongoARMtemplate%2Fmaster%2Fazuredeploy.json" target="_blank">
+    <img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a>
+<a href="
+http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fcjsingh8512%2Fazure-cosmosdb-mongoARMtemplate%2Fmaster%2Fazuredeploy.json" target="_blank">
+    <img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.png"/>
+</a>
+
+
+This template deploys a MongoDB Sharding Cluster on CentOS. It deploys 1 router server, one config server replica set with 3 nodes, and 1 shard which is a replica set with 4 nodes. So it totally deploys 8 nodes.
+
+The router server is exposed on public IP address that you can access through SSH on the standard port, also mongodb port 27017 open. You can access it for MongoDB data write and read.
+
+The config server replica set stores sharding cluster metadata. MongoDB suggests to use a replica set for the metadata store in the production environment, in case one of the config server is down, there will still be other 2 config server nodes offer the service.
+
+1 shard is a 4 node replica set. You can shard the data on the replica set. You can also add more replica sets into the sharding cluster.
+
+The nodes are under the same subnet 10.0.0.0/24. Except the router server, the other nodes only have private IP address.
+
+<img src="https://raw.githubusercontent.com/cjsingh8512/azure-cosmosdb-mongoARMtemplate/master/images/Mongo Sharded Cluster.png" />
```

Comment:
Diagram needs a few cosmetic updates: Align the boxes, move labels into the rectangle instead of having a separate label underneath.

Reply:
Done